persistence/trees/trees.go

```diff
@@ -78,12 +79,17 @@ const (
 	numMerkleTrees
 )
 
+// Ensure treeStore implements TreeStore
+var _ modules.TreeStoreModule = &treeStore{}
+
 // treeStore stores a set of merkle trees that
 // it manages. It fulfills the modules.TreeStore interface.
 // * It is responsible for atomic commit or rollback behavior
 // of the underlying trees by utilizing the lazy loading
 // functionality provided by the underlying smt library.
 type treeStore struct {
```

Comment:
@dylanlott Wdyt of adding a field here such as `rootTree *smt.SMT` which contains the different statetree name-hash pairings. This way we no longer need to keep track of the order of the trees we just insert their root into this tree and this trees root is our stateHash. Updates in any order to the SMT will produce the same tree.

Reply:
Root store doesn't need to not hash values as it will simply be used for 1) root hash calculation and 2) inclusion proofs of a subtree in the state
 - the IBC state tree will have no value hashing which is where the difference will be but this is still the same `*smt.SMT` type and poses no real changes except for how the IBC state tree will be made
 
 As the root store is going to be made of the hashes of the sub trees - it makes sense to me to use the `KVStore` Postgres seems overkill. It could even be an in memory `smt.SimpleMap` as the main purpose will be root calculation - and the only other thing that will use it is the IBC light client verification system - this will build the tree using the hashes of the sub trees which will be added to the block header. So in reality an in memory ephemeral tree is probably best
 
The benefits of this approach is that we no longer have to track the order of the trees as the SMT is not order dependent for its insertions

Reply:
> The benefits of this approach is that we no longer have to track the order of the trees as the SMT is not order dependent for its insertions

 This will be good, the order dependent var list of trees was an eye sore and a half 💩

Reply:
Infact it doesnt even need a node store

```go
lazy := smt.NewSparseMerkleTree(nil, sha256.New())
_ = lazy.Update([]byte("foo"), []byte("oof"))
_ = lazy.Update([]byte("bar"), []byte("rab"))
_ = lazy.Update([]byte("baz"), []byte("zab"))
lazyRoot := lazy.Root()
fmt.Println(lazyRoot)

lazy = smt.NewSparseMerkleTree(nil, sha256.New())
_ = lazy.Update([]byte("baz"), []byte("zab"))
_ = lazy.Update([]byte("bar"), []byte("rab"))
_ = lazy.Update([]byte("foo"), []byte("oof"))
lazyRoot = lazy.Root()
fmt.Println(lazyRoot)
```
```
[194 210 162 176 162 41 147 217 75 19 160 224 30 159 210 181 33 239 178 208 203 236 240 248 145 48 127 62 53 46 139 169]
[194 210 162 176 162 41 147 217 75 19 160 224 30 159 210 181 33 239 178 208 203 236 240 248 145 48 127 62 53 46 139 169]
```

Reply:
Done in #843